### PR TITLE
Push logs to logstash

### PR DESCRIPTION
**Pull request by hieunguyen23**
I updated branch push-logs-to-logstash. Please review the code for me. 
Thanks Mr.@tuanta.

### CODE REVIEW

File: zimbra_all/opt/push-logs-to-logstash/02-output.conf [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+# input {
+#   udp {
+#     host => "logstash_ip"
+#     port => 10514
+#     codec => "json"
+#     type => "rsyslog"
+#   }
+# }
+
```
Comment (tuanta) [inline zimbra_all/opt/push-logs-to-logstash/02-output.conf:18]: What does this block mean?

File: zimbra_all/opt/build.sh [hunk 2]
```diff
@@ -20,6 +20,17 @@
...
+then
+      echo "Push log to logstash not define, so not config"
+      rm -rf /opt/push-logs-to-logstash
+else
+      echo "Push log to logstash server $IP_LOGSTASH, configuring...."
+      cp /opt/push-logs-to-logstash/* /etc/rsyslog.d/
+      sed -i "s|IP_LOGSTASH|$IP_LOGSTASH|g" /etc/rsyslog.d/02-output.conf
+      rm -rf /opt/push-logs-to-logstash
+fi
+
```
Comment (tuanta) [inline zimbra_all/opt/build.sh:33]: Parameter name should be LOGSTASH_IP.

**Review by tuanta — CHANGES_REQUESTED** [review]
@hieunguyen23 : please check my reviews.

File: zimbra_all/opt/push-logs-to-logstash/02-output.conf (hunk 1 above)
Comment (hieunguyen23) [reply zimbra_all/opt/push-logs-to-logstash/02-output.conf:18]: It is used as a hint to configure logstash to listen to port 10514/udp.

File: README.md [hunk 3]
```diff
@@ -35,5 +35,18 @@ $ docker run --name zimbra-iwaytest2 -it \
...
+##### Run a new Zimbra container with push logs to logstash (optional):
+```bash
+$ docker run --name zimbra-iwaytest2 -it \
+-p 25:25 -p 80:80 -p 465:465 \
+-p 587:587 -p 110:110 -p 143:143 \
+-p 993:993 -p 995:995 -p 443:443 \
+-p 3443:3443 -p 9071:9071 \
+-h mail.iwaytest2.com --net=zimbranet --dns 172.28.0.1 \
+-v zimbra-iwaytest2:/opt/zimbra \
+-e PASSWORD=Zimbra2021 -e IP_LOGSTASH=192.168.100.252 iwayvietnam/zimbra_all
```
[the remark this answers is not in the document]
Comment (hieunguyen23) [reply README.md]: I fixed it, thank you

File: docker-compose/docker-compose.yml [hunk 4]
```diff
@@ -50,6 +50,7 @@ services:
           - './zimbra-storage:/opt/zimbra:rw'
       environment:
           - PASSWORD=Zimbra2021
+          #- IP_LOGSTASH=192.168.100.252 #if you want push logs to logstash (optional)
```
[the remark this answers is not in the document]
Comment (hieunguyen23) [reply docker-compose/docker-compose.yml:53]: I fixed it, thank you

File: zimbra_all/opt/build.sh (hunk 2 above)
Comment (hieunguyen23) [reply zimbra_all/opt/build.sh]: I fixed it, thank you

**Pull request closed without merging**